frontends/mit-open/src/pages/FieldPage/EditFieldAppearanceForm.tsx

```diff
@@ -1,47 +1,73 @@
 import React, { useCallback } from "react"
 import { useNavigate } from "react-router-dom"
 import { useFormik } from "formik"
-import { Button, TextField } from "ol-components"
+import { RadioChoiceField, Button, TextField } from "ol-components"
 import * as Yup from "yup"
 
 import type { FieldChannel } from "api/v0"
 import { makeFieldViewPath } from "@/common/urls"
-import { useFieldPartialUpdate } from "api/hooks/fields"
+import { useChannelPartialUpdate } from "api/hooks/fields"
 
 type FormProps = {
   field: FieldChannel
 }
-
+const CHANNEL_TYPE_CHOICES = [
+  {
+    value: "department",
+    label: "Department",
+    className: "radio-option",
+  },
+  {
+    value: "topic",
+    label: "Topic",
+    className: "radio-option",
+  },
+  {
+    value: "offeror",
+    label: "Offeror",
+    className: "radio-option",
+  },
+  {
+    value: "pathway",
+    label: "Pathway",
+    className: "radio-option",
+  },
+]
 const postSchema = Yup.object().shape({
   title: Yup.string().default("").required("Title is required."),
   public_description: Yup.string()
     .default("")
     .required("Description is required."),
+  channel_type: Yup.string()
+    .default("pathway")
+    .required("Channel Type is required."),
 })
 type FormData = Yup.InferType<typeof postSchema>
 
 const EditFieldAppearanceForm = (props: FormProps): JSX.Element => {
   const { field } = props
-  const fieldName = field.name
-  const editField = useFieldPartialUpdate()
+  const fieldId = field.id
+  const editField = useChannelPartialUpdate()
   const navigate = useNavigate()
 
   const handleSubmit = useCallback(
     async (e: FormData) => {
-      const data = await editField.mutateAsync({ field_name: fieldName, ...e })
+      // @ts-expect-error No idea how to fix this
+      const data = await editField.mutateAsync({ id: fieldId, ...e })
```

Comment:
@ChristopherChudzicki any idea how to get rid of this error message? For the time being I gave up and added `@ts-expect-error` for the line so it gets ignored.


> ➤ YN0000: [mit-open]: src/pages/FieldPage/EditFieldAppearanceForm.tsx(55,48): error TS2345: Argument of type '{ title: string; public_description: string; channel_type: string; id: number; }' is not assignable to parameter of type 'PatchedFieldChannelWriteRequest & { id: number; }'.
> ➤ YN0000: [mit-open]:   Type '{ title: string; public_description: string; channel_type: string; id: number; }' is not assignable to type 'PatchedFieldChannelWriteRequest'.
> ➤ YN0000: [mit-open]:     Types of property 'channel_type' are incompatible.
> ➤ YN0000: [mit-open]:       Type 'string' is not assignable to type 'ChannelTypeAe2Enum | undefined'.
> ➤ YN0000: [mit-open]: Process exited (exit code 2), completed in 15s 740ms
> 

Relevant code from the autogenerated schema:

```
export const ChannelTypeAe2Enum = {
  Topic: "topic",
  Department: "department",
  Offeror: "offeror",
  Pathway: "pathway",
} as const

export type ChannelTypeAe2Enum =
  (typeof ChannelTypeAe2Enum)[keyof typeof ChannelTypeAe2Enum]
  ```

Reply:
_Aside: I'm curious why DRF-Spectacular is making two copies of ChannelTypeAe2Enum ... Do you have any idea? It doesn't really matter—they are the same._

To the typescript error:
- `mutateAsync` knows that `channel_type` should be one of the specific values in ChannelTypeAe2Enum
- but in `async (e: FormData) => {...}`, FormData only knows that channel_type is a string. 

The `FormData` type is coming from the `yup` schema. The best resolution is
```diff
@@ -39,6 +39,7 @@ const postSchema = Yup.object().shape({
     .default("")
     .required("Description is required."),
   channel_type: Yup.string()
+    .oneOf(Object.values(ChannelTypeEnum))  // or ChannelTypeAe2Enum, they are the same
     .default("pathway")
     .required("Channel Type is required."),
 })
@@ -52,7 +53,6 @@ const EditFieldAppearanceForm = (props: FormProps): JSX.Element => {

   const handleSubmit = useCallback(
     async (e: FormData) => {
-      // @ts-expect-error No idea how to fix this
       const data = await editField.mutateAsync({ id: fieldId, ...e })
       if (data) {
         navigate(makeFieldViewPath(String(data.channel_type), data.name))
@@ -67,7 +67,7 @@ const EditFieldAppearanceForm = (props: FormProps): JSX.Element => {
     initialValues: {
       title: field.title,
       public_description: String(field.public_description),
-      channel_type: String(field.channel_type),
+      channel_type: field.channel_type,
     },
     validationSchema: postSchema,
     onSubmit: handleSubmit,
```
Then `yup` knows it should be one of the specific values.

**NOTES:**
- I then had to remove `String(...)` from the initial values, otherwise TS will think channel type is any string there (and now yup knows it should be a specific..)

Reply:
Incidentally, Carey ran into this exact issue yesterday.

Reply:
Thanks!